dev/com.ibm.ws.jaxws.2.2.webcontainer_fat_extended/fat/src/com/ibm/ws/jaxws/fat/IgnoreUnexpectedElementConfigTest.java

```diff
@@ -0,0 +1,287 @@
+        LOG.info("testServiceNameIgnoreUnexpectedElementTrue - Response = " + response);
+
+        assertTrue("Expected successful response, but response was " + response,
+                   response.contains("Hello,AddedElement"));
+    }
+
+    /**
+     * Test the serviceName and global configs with mis-mataching ignoreUnknownElement elements in server.xml
+     *
+     * Server config - servicename-false-global-true-server.xml
+     * Expected response - Unmarshalling Error: unexpected element
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testServiceNameFalseAndGlobalTrueIgnoreUnexpectedElement() throws Exception {
+        addedElementServer.reconfigureServer("AddedElementHandlerChainTestServer/servicename-false-global-false-server.xml", "CWWKG0017I");
+
+        // Visit the URL:  http://hostName:hostPort/testHandlerClient/TestServiceServlet?target=user
+        String response = runTest(addedElementServer, "testHandlerClient/IgnoreUnexpectedElementTestServiceServlet?target=AddedElement");
+        // Log response to output.txt
+        LOG.info("testServiceNameIgnoreUnexpectedElementFalse - Response = " + response);
+
+        assertNull("Expected null response from server, but was" + response, response);
+        assertNotNull("Expected Unmarshalling Error: unexpected element exception is not thrown",
+                      addedElementServer.waitForStringInLog("xml.ws.soap.SOAPFaultException: Unmarshalling Error: unexpected element "));
+    }
+
+    /**
+     * Test an incorrect serviceName attribute doesn't apply the config for ignoreUnexpectedElements
+     *
+     * Server config - incorrect-servicename-server.xml
+     * Expected response - Unmarshalling Error: unexpected element
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testIncorrectServiceNameIgnoreUnexpectedElement() throws Exception {
+        addedElementServer.reconfigureServer("AddedElementHandlerChainTestServer/incorrect-servicename-server.xml", "CWWKG0017I");
+
+        // Visit the URL:  http://hostName:hostPort/testHandlerClient/TestServiceServlet?target=user
+        String response = runTest(addedElementServer, "testHandlerClient/IgnoreUnexpectedElementTestServiceServlet?target=AddedElement");
+        // Log response to output.txt
+        LOG.info("testServiceNameIgnoreUnexpectedElementFalse - Response = " + response);
+
+        assertNull("Expected null response from server, but was" + response, response);
+        assertNotNull("Expected Unmarshalling Error: unexpected element exception is not thrown",
+                      addedElementServer.waitForStringInLog("xml.ws.soap.SOAPFaultException: Unmarshalling Error: unexpected element "));
+    }
+
+    private String runTest(LibertyServer server, String pathAndParams) throws ProtocolException, IOException {
+        URL url = new URL("http://" + server.getHostname() + ":" + server.getHttpDefaultPort() + "/" + pathAndParams);
```

Comment:
Adding a wait for server configuration update might be good.

Reply:
It is good for sure, that's actually why at the begining of each test I used this method: 

`ddedElementServer.reconfigureServer("AddedElementHandlerChainTestServer/global-true-server.xml", "CWWKG0017I");` 

It automatically waits for the the server reconfig message in the logs - `"CWWKG0017I"`